src/agent/onefuzz/src/blob/url.rs

```diff
@@ -1,24 +1,28 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-use std::fmt;
+use std::{fmt, path::PathBuf};
 
 use anyhow::Result;
 use reqwest::Url;
 use serde::{de, Serialize, Serializer};
 
 #[derive(Clone, Eq, PartialEq)]
-pub struct BlobUrl {
-    url: Url,
+pub enum BlobUrl {
```

Comment:
The goal of this abstraction was to be a `Url`, but one that we know to follow the default Azure blob storage pattern, which guarantees that we can extract a container and account name.

I would have liked if we kept it as-is, but then added a _new_ `enum` that can be either/or, as in:
```rust
pub enum Blob {
    Remote(BlobUrl),
    Local(PathBuf),
}
```

Reply:
I agree however this will be a big change as it might involve doing the same thing to the blobContainer abstraction
This will be addressed in a different PR. I created #670 to keep track of this